### PR TITLE
NETOBSERV-1201: Make empty CR work out of the box

### PR DESCRIPTION
**Pull request by jotak**
Fixing a couple of default values that weren't working as expected:
- metav1.Duration fields need to be pointers so their default is actually used
- deploymentModel shouldn't be required as it default to DIRECT
- Agent type default wasn't working due to missing omitempty
- Add a test to make sure empty spec works

### CODE REVIEW

File: pkg/helper/helpers.go [hunk 1]
```diff
@@ -68,3 +70,10 @@ func MaxLabelLength(in string) string {
 	}
 	return in[:maxLabelLength]
 }
+
+func UnstructuredDuration(in *metav1.Duration) string {
```
Comment (msherif1234) [inline pkg/helper/helpers.go:74]: pls add ut coverage for this api, I have one for u :)
```golang
func TestUnstructuredDuration(t *testing.T) {

  t.Run("nil input", func(t *testing.T) {
    var d *metav1.Duration
    got := UnstructuredDuration(d)
    want := ""
    
    if got != want {
      t.Errorf("got %q, want %q", got, want) 
    }
  })

  t.Run("valid input", func(t *testing.T) {
    d := &metav1.Duration{Duration: time.Minute}
    want := "1m0s"
    
    got := UnstructuredDuration(d)
    
    if got != want {
      t.Errorf("got %q, want %q", got, want)
    }
  })

}
```

**Review by msherif1234 — COMMENTED** [review]
/LGTM
just suggested adding UT coverage and u need to fix rebase issues

File: pkg/helper/helpers.go (hunk 1 above)
Comment (jotak) [reply pkg/helper/helpers.go:74]: thanks! done

**Comment by jotak** [comment]
Merging for dequeuing - I don't think specific tests are needed here, apart from usual regression tests
/approve
